tests/desktop_gnome_contacts.pm

```diff
@@ -0,0 +1,90 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    check_desktop_clean();
+
+    # Run Contacts.
+    type_very_safely("gnome-contacts\n");
+    assert_screen('gnome_contacts_setup', 10);
+
+    # Go through the initial setup, selecting Local Address Book.
+    # FIXME: Some improved keyboard navigation for this would be good.
+    # See https://gitlab.gnome.org/GNOME/gnome-contacts/issues/130.
+    send_key('tab');  # source list
+    send_key('spc');  # select ‘Local Address Book’
+    send_key('tab');  # ‘Quit’ button
+    send_key('tab');  # ‘Done’ button
+    send_key('ret');  # press the button
+
+    # Force gnome-contacts to be maximised, as this makes the needles a little
+    # simpler.
+    send_key_combo('super', 'up');
```

Comment:
It's being launched through desktop search and should be automatically maximized like (almost) every other app on Endless – is it not?

Reply:
Sort of. I was seeing the initial window (the welcome wizard) maximised, then the main window (which appears after the wizard is done) would be un-maximised, and then the main window would be maximised again when you re-launch gnome-contacts (and don’t go through the wizard because you’ve done that already).

So I guess what’s happening is:
 1. Wizard is launched, EOS shell maximises it.
 2. Main window is shown, gnome-contacts restores the saved maximisation state from GSettings, which at this point is the default of `false`. Main window is un-maximised.
 3. App is re-launched, EOS shell maximises the main window this time.

So if we want to fix this papercut, and assuming my guesses are correct (I haven’t checked the gnome-contacts code), we could either:
 1. Make the GSettings saved maximisation state apply to the wizard, and also to be updated from the wizard, so that EOS shell maximising the wizard results in the GSettings value being updated to `true`.
 2. Change the default value of the GSettings maximisation state to `true`.

Option 1 seems better, since it would actually be upstreamable.

Reply:
I looked at the code and my assumptions seem to fit. I’ve filed https://gitlab.gnome.org/GNOME/gnome-contacts/issues/151, but don’t plan on fixing it (because I don’t have a working checkout of gnome-contacts at the moment and getting it to build is a bit of a mission).